src/main/java/org/broadinstitute/hellbender/tools/walkers/GenotypeGVCFsEngine.java

```diff
@@ -183,21 +185,37 @@ private VariantContext regenotypeVC(final VariantContext originalVC, final Refer
      * Remove NON-REF alleles from the variant context
      *
      * @param vc   the variant context
+     * @param infoHeaderAltAllelesLineNames INFO Header names that require alt alleles, which will be removed if the ALT alleles change.
+     * @param removeUnusedAlternates If true, unused alternate alleles will also be trimmed.
      * @return variant context with the NON-REF alleles removed if multiallelic or replaced with NO-CALL alleles if biallelic
      */
-    private VariantContext removeNonRefAlleles(final VariantContext vc) {
+    @VisibleForTesting
+    static VariantContext removeNonRefAlleles(final VariantContext vc, final Set<String> infoHeaderAltAllelesLineNames, final boolean removeUnusedAlternates) {
 
         // If NON_REF is the only alt allele, ignore this site
         final List<Allele> newAlleles = new ArrayList<>();
         // Only keep alleles that are not NON-REF
+        boolean altAllelesChanged = false;
         for ( final Allele allele : vc.getAlleles() ) {
             if ( !allele.equals(Allele.NON_REF_ALLELE) ) {
                 newAlleles.add(allele);
             }
+            else {
+                altAllelesChanged = true;
+            }
+        }
```

Comment:
Can we simplify the bulk of this method to just
```
final Set<Allele> allelesToKeep = removeUnusedAlternates ? vc.getGenotypes().stream().flatMap(g->g.getAlleles().stream()).collect(Collectors.toSet()) : vc.getAlleles();
allelesToKeep.remove(Allele.NON_REF_ALLELE);
final List<Allele> newAlleles = vc.getAlleles().stream().filter(allelesToKeep::contains).collect(Collectors.toList());
```

Reply:
I modified this slightly, but yes something like this is simpler than what's there.  The main difference is to ensure REF is retained